zbus/tests/e2e.rs

```diff
@@ -266,7 +266,7 @@ impl MyIface {
     #[zbus(property)]
     fn test_header_prop(
         &self,
-        #[zbus(header)] header: Option<Header<'_>>,
+        #[zbus(header)] header: Option<&Header<'_>>,
```

Comment:
This is inconsistent now with the `header` arg that normal methods get. I think `Option<Header>` is fine but if we want to change this, we need to also change the normal methods' arg type to `&Header`.

Reply:
Is `Option<&Header>` not what you wanted? That's what I believe you asked for on Matrix.

![Screenshot From 2024-12-22 13-58-08](https://github.com/user-attachments/assets/1d756991-57b9-47cb-910a-1dedfb8e5eca)

I also did notice that while the other _method_ special args are being passed by reference, `Header` wasn't in a Matrix comment:

![Screenshot From 2024-12-22 13-59-44](https://github.com/user-attachments/assets/2473cee3-a90e-45f7-9d58-293f9128f3fe)

I'm sorry, but I'm slightly confused. What do you want we do here? Change `Option<&Header>` to `Option<Header>` (in spite of the original request for `Option<&T>`)? Or changing the old parameters to take `&Header` instead of `Header`?

Reply:
As I wrote there, I was talking about the `&Option<Header>` arg in the `Interface` property methods. They should be changed to `Option<&Header>`. I also wrote:

> ah that's only the Interface methods but still we should change it

i-e I didn't suggest we change anything in the `interface` macro args since they're not doing `&Option<..>` but `Option<..>` (i-e there is no reference involved).

Reply:
> As I wrote there, I was talking about the `&Option<Header>` arg in the `Interface` property methods.

OK, I'll see about re-spinning this then.